vercel/turbo
# feat: add ability to spawn child process with pty

### Description

This PR adds, but doesn't hook up, the ability to spawn a child process hooked up to a pseudo terminal.

I *highly* suggest reviewing each commit on its own. I tried my best to split out prefactoring work from any feature work.

### Testing Instructions

Adapt unit tests to cover all of them. It will be noted that switching to use a PTY does affect what programs output and so the tests were made to not check for identical output, but just that we got the expected output.
Closes TURBO-2128
